giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/controllers/FileContentController.java

```diff
@@ -39,7 +44,7 @@
     public ResponseEntity<String> getFile(
             @PathVariable String fileId,
             HttpServletResponse response,
-            HttpServletRequest request, Principal principal) {
+            HttpServletRequest request, Principal principal) throws NoNepomukFoundException {
```

Comment:
where is this thrown? Since this is a controller, you need to handle the exception appropriately. Whoever calls this api endpoint needs to get a proper response.